src/htmx.js

```diff
@@ -3603,13 +3603,22 @@ return (function () {
         //====================================================================
         // Initialization
         //====================================================================
+        var isReady = false
+        getDocument().addEventListener('DOMContentLoaded', function() {
+          isReady = true
+        })
 
+        /**
+         * Execute a function now if the DOM content has been loaded, otherwise execute it when DOM content is loaded.
+         *
+         * This function uses isReady because there is no realiable way to ask the browswer whether the DOMContentLoaded event
+         * has already been fired; there's a gap between `DOMContentLoaded` firing and readystate=complete.
+         */
         function ready(fn) {
-            if (getDocument().readyState === 'complete') {
-                fn();
-            } else {
-                getDocument().addEventListener('DOMContentLoaded', fn);
-            }
+          // Call the function now if DOMContentLoaded has been fired
+          if (isReady || getDocument().readyState === 'complete') return fn()
```

Comment:
Is this really better? Why not keep if/else?

Reply:
On second read though, I sort of see your point. When I wrote it I just nuked the whole function but I could write that in a more diff-minimizing way (which I try to do where reasonable).